python/lsst/daf/butler/remote_butler/_remote_butler.py

```diff
@@ -80,8 +94,15 @@ def __init__(
             # This is generally done for testing.
             self._client = http_client
         else:
-            headers = {"user-agent": f"{get_full_type_name(self)}/{__version__}"}
-            self._client = httpx.Client(headers=headers, base_url=str(self._config.remote_butler.url))
+            server_url = str(self._config.remote_butler.url)
+            auth_headers = {}
+            if access_token is None:
+                access_token = get_authentication_token_from_environment(server_url)
+            if access_token is not None:
+                auth_headers = get_authentication_headers(access_token)
+
+            headers = auth_headers | {"user-agent": f"{get_full_type_name(self)}/{__version__}"}
```

Comment:
Cute. Alternatively:

```python
headers = {"user-agent": f"{get_full_type_name(self)}/{__version__}"}
headers.update(auth_headers)
```
Which might be clearer eventually if we are ever going to put more than `user-agent` in the second dict above.